src/db.rs

```diff
@@ -121,22 +184,20 @@ pub fn get_connection(uuid: Uuid) -> Result<Connection> {
 }
 
 pub fn get_secret_policy(sec: &str) -> Option<policy::Policy> {
-    let mut dbconn = get_dbconn().unwrap();
+    let mut dbconn = get_dbconn().ok()?;
     let mut trnsx = dbconn.start_transaction(TxOpts::default()).ok()?;
 
-    let mqstr = "SELECT polids FROM secpol WHERE secret = ?";
-
-    let val: Option<i32> = trnsx.exec_first(mqstr, (sec,)).ok()?;
+    let mqstr = "SELECT polid FROM secrets WHERE secret_id = ?";
 
-    if val.is_none() {
-        info!("No policy has been set for secret with id: {}", &sec);
-        return None;
-    }
+    let val_row: mysql::Row = trnsx.exec_first(mqstr, (sec,)).ok()?.unwrap();
+    let val = match mysql::from_value_opt::<u64>(val_row["polid"].clone()) {
+        Ok(u) => u,
+        Err(_e) => return None,
+    };
 
     let mqstr = "SELECT allowed_digests, allowed_policies, min_fw_api_major, min_fw_api_minor, allowed_build_ids FROM policy WHERE id = ?";
```

Comment:
this SELECT also appears in get_policy (but there with a `AND valid=1` clause). can this function call get_policy here?

Also the buildup of the polic::Policy struct below is a lot cleaner here than in get_policy. Why is there a difference?

Reply:
Yes, fixed to call get_policy after the val match.

Reply:
Sorry, hit comment button too soon.  The difference is that I was trying to use mysql::Row to use the , but got lost in the weeds.  I have updated get_policy to use the same vector approach.

Reply:
Looks OK. Now please remove the commented-out code.